src/main/java/com/google/sps/servlets/SpecificLocationEventsServlet.java

```diff
@@ -0,0 +1,121 @@
+package com.google.sps.servlets;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Collections;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.sps.data.Constants;
+import com.google.sps.data.Event;
+import com.google.gson.Gson;
+import java.lang.*;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.TimeZone;
+import java.text.ParseException; 
+
+@WebServlet("/location-specific-events")
+public class SpecificLocationEventsServlet extends HttpServlet {
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
```

Comment:
I'd argue this should be a `doGet`. The action fetches a resource and doesn't do any mutations.

Reply:
I agree, initially I had written it as a doGet. However, I do not know how to pass parameters to a doGet function, and I need to pass the placeID param to the function. Is there a way to pass parameters to a doGet?